components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -518,14 +519,17 @@ private boolean canCountTotalUserCount(String[] userStoreDomainNames) {
     private long getTotalUsers(String domainName) throws CharonException {
 
         long totalUsers = 0;
-        if (carbonUM instanceof JDBCUserStoreManager) {
-            try {
-                AbstractUserStoreManager secondaryUserStoreManager = (AbstractUserStoreManager) carbonUM
-                        .getSecondaryUserStoreManager(domainName);
-                totalUsers = secondaryUserStoreManager.countUsersWithClaims("http://wso2.org/claims/username", "*");
-            } catch (org.wso2.carbon.user.core.UserStoreException e) {
-                throw new CharonException("Error while getting total user count in domain: " + domainName);
+        AbstractUserStoreManager secondaryUserStoreManager = null;
+        if (StringUtils.isNotBlank(domainName)) {
+            secondaryUserStoreManager = (AbstractUserStoreManager) carbonUM
+                    .getSecondaryUserStoreManager(domainName);
+        }
+        try {
+            if (secondaryUserStoreManager instanceof JDBCUserStoreManager) {
+                totalUsers = secondaryUserStoreManager.countUsersWithClaims(USERNAME_CLAIM, "*");
             }
+        } catch (org.wso2.carbon.user.core.UserStoreException e) {
+            throw new CharonException("Error while getting total user count in domain: " + domainName);
```

Comment:
By doing this we lose the original stack trace of the UserStoreException

Reply:
https://github.com/wso2-extensions/identity-outbound-auth-email-otp/pull/79/files#diff-fd41cbd384579a5bdf1bdf1b6c8f4463R131